core/tests/webdriverio_utils/ExplorationEditorSettingsTab.js

```diff
@@ -26,15 +26,19 @@ var ExplorationEditorSettingsTab = function() {
   /*
    * Interactive elements
    */
-  var explorationCategoryInput = $('.e2e-test-exploration-category-input');
+  var explorationCategory = $('.e2e-test-exploration-category');
   var explorationLanguageInput = $('.e2e-test-exploration-language-select');
   var explorationObjectiveInput = $('.e2e-test-exploration-objective-input');
   var explorationObjectiveWarning = $(
     '.e2e-test-exploration-objective-warning');
   var explorationSummaryTile = $('.e2e-test-exploration-summary-tile');
   var explorationTitleInput = $('.e2e-test-exploration-title-input');
   var initialStateSelect = $('.e2e-test-initial-state-select');
-  var initialStateSelector = '.e2e-test-initial-state-select';
+  var initialStateSelectOption = function(stateName) {
+    return $(`.e2e-test-initial-state-select-element=${stateName}`);
+  };
```

Comment:
Why are you using single selectors in a function? Only all selectors are being referenced using functions for the e2e migration.  /cc @ShivamJhaa.

Reply:
this was introduced in PR #15836

Reply:
@ShivamJhaa PTAL

Reply:
We are using the same way in protractor code as well PTAL [here](https://github.com/oppia/oppia/blob/develop/core/tests/protractor_utils/ExplorationEditorSettingsTab.js#L43). `initialStateSelectOption` selects element element on the basis of stateName provided by the test suites, so in order to reduce code duplication we are making a function.